astropy/units/structured.py

```diff
@@ -0,0 +1,410 @@
+
+        Returns
+        -------
+        bool
+        """
+        if not isinstance(other, type(self)):
+            try:
+                other = self.__class__(other, self.names)
+            except Exception:
+                return False
+
+        if len(self) != len(other):
+            return False
+
+        for self_part, other_part in zip(self.values(), other.values()):
+            if not self_part.is_equivalent(other_part,
+                                           equivalencies=equivalencies):
+                return False
+
+        return True
+
+    def _get_converter(self, other, equivalencies=[]):
+        if not isinstance(other, type(self)):
+            other = self.__class__(other, self.names)
+
+        converters = [self_part._get_converter(other_part,
+                                               equivalencies=equivalencies)
+                      for (self_part, other_part) in zip(self.values(),
+                                                         other.values())]
+
+        def converter(value):
+            if not hasattr(value, 'dtype'):
+                value = np.array(value, self._recursively_get_dtype(value))
+            result = np.empty_like(value)
+            for name, converter_ in zip(result.dtype.names, converters):
+                result[name] = converter_(value[name])
+            return result
+
+        return converter
+
+    def to(self, other, value, equivalencies=[]):
+        """Return values converted to the specified unit.
+
+        Parameters
+        ----------
+        other : `~astropy.units.StructuredUnit`
+            The unit to convert to.  If necessary, will be converted to
+            a Structured Unit using the dtype of ``value``.
+        value : array-like
+            Value(s) in the current unit to be converted to the
+            specified unit.  If a sequence, the first element must have
+            entries of the correct type to represent all elements (i.e.,
+            not have, e.g., a ``float`` where other elements have ``complex``).
+        equivalencies : list of tuple, optional
+            A list of equivalence pairs to try if the units are not
+            directly convertible.  See :ref:`unit_equivalencies`.
+            This list is in addition to possible global defaults set by, e.g.,
+            `set_enabled_equivalencies`.
+            Use `None` to turn off all equivalencies.
+
+        Returns
+        -------
+        values : scalar or array
+            Converted value(s).
+
+        Raises
+        ------
+        UnitsError
+            If units are inconsistent
+        """
+        return self._get_converter(other, equivalencies=equivalencies)(value)
+
+    def to_string(self, format='generic'):
+        """Output the unit in the given format as a string.
+
+        Units are separated by commas.
+
+        Parameters
+        ----------
+        format : `astropy.units.format.Base` instance or str
```

Comment:
Currently it only supports strings.

Reply:
Actually, not quite, and also it works for all classes. I've rewritten this.